### ISSUE TITLE
Transitions with dynamic targets

### ISSUE THREAD

**Issue by xcambar** [pzuraq/liquid-tether#63]
Hi,
I'm building an onboarding tour for an application, trying to use the liquid-tether transitions with the following template:

```hbs
{{#liquid-tether
  to="tour"
  isFirstStep=isFirstStep
  isLastStep=isLastStep
  target=currentStep.element
  attachment=currentStep.attachment
  targetAttachment=currentStep.targetAttachment
  }}
  <!-- Too many HTML -->
{{/liquid-tether}}
```

and the following `transition.js`

```js
  this.transition(
    target('tour'),
    this.toValue(({ isFirstStep, isLastStep })=> !!isFirstStep || !!isLastStep),
    this.use('tether', ['fade', { duration: 250 }])
  );
  this.transition(
    target('tour'),
    this.toValue(({ isFirstStep, isLastStep })=> !isFirstStep && !isLastStep),
    this.use('tether', ['fly-to', { duration: 500 }])
  );
```

As you can see in the template above, the target is bound to a variable, contrary to all the examples you provide, in which you use a `showFirstFlytoDialog`, `showSecondFlytoDialog` (which I'm trying to avoid as much as possible).

What I'm seeing though, is that the first step of my onboarding is well animated (`fade` in), but when the user moves to step 2 and beyond, there's no animation whatsoever.

If you need more info, I'll be happy to answer your questions.

**tl;dr;** Can I build something similar to the Animated modals example with Liquid Tether with a single `liquid-tether` and a bound variable as the (moving) target?

**Comment by pzuraq** [comment]
I like the idea and the API! Very clean and simple. I'll see if I can make this happen, working on v2 right now so it should be possible, assuming I can figure out how to make tether behave.

**Comment by xcambar** [comment]
Thanks for the comment, very appreciated 😄 .
As I realise it does not seem to be a feature _yet_, I'll try to have a look as well.

**Comment by pzuraq** [comment]
So it looks like this is going to require a fairly large change to the API for `toValue` and `fromValue`. Currently you can't really transition from a wormhole to itself, liquid fire just doesn't recognize that there was a change so no transition occurs. I'm going to dig into this, would like to get it shipped in v2.

**Comment by pzuraq** [comment]
@xcambar So the way I'm thinking we can solve this is by adding a new parameter to the component, `meta`. This field would have to be an object, and it would contain all of the values you want to match on in your transitions.

So, your example above would become:

```
{{#liquid-tether
  to="tour"
  meta=(hash isFirstStep=isFirstStep isLastStep=isLastStep)
  target=currentStep.element
  attachment=currentStep.attachment
  targetAttachment=currentStep.targetAttachment
  }}
  <!-- Too many HTML -->
{{/liquid-tether}}
```

This is something I've been thinking about doing for a while, since passing in the entire component to the `toValue`/`fromValue` helpers seemed like a little bit much, but this is the first time it would actually enable new functionality.

**Comment by pzuraq** [comment]
Actually, we can do it without it being an object. It can be whatever value you want it to be. We could even name it `value` so our helpers actually make sense.

**Comment by pzuraq** [comment]
This is solved by liquid-tether 2 beta 2! Thanks again for the suggestion, I think it works wonderfully 👍

**Comment by xcambar** [comment]
It does indeed, congrats!

Great new doc, by the way.
